src/app/content/components/Content.tsx

```diff
@@ -17,25 +17,39 @@ interface PropTypes {
 
 interface ReactState {
   book?: ArchiveContent;
-  page?: ArchiveContent;
+  page?: ArchivePage;
 }
 
 export class ContentComponent extends Component<PropTypes, ReactState> {
   public state: ReactState = {};
 
+  constructor(props: PropTypes) {
+    super(props);
+    const {book, page, services} = props;
+
+    if (book) {
+      this.state.book = services.archiveLoader.cachedBook(book.shortId);
+    }
+    if (page && book) {
+      this.state.page = services.archiveLoader.cachedPage(book.shortId, page.shortId);
+    }
+  }
+
   public loadBook(props: PropTypes) {
-    if (!props.book) {
+    const {book, services} = props;
+    if (!book) {
       return;
     }
-    this.props.services.archiveLoader(props.book.shortId).then((book) => this.setState({book}));
+    this.setState({book: services.archiveLoader.cachedBook(book.shortId)});
```

Comment:
Why does the `cachedBook` method now need to be called rather than just the `archiveLoader` function?

Reply:
this is done in the react component to avoid extraneous promises that mess with the server side render